store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -406,14 +406,12 @@ public void setupEditHeaderData(Arguments arguments, AbstractCommand ac) throws
      */
     public boolean matchCondition(ZimbraMailAdapter mailAdapter, Header header, List<String> headerList, String value, SieveContext context) throws LookupException, SieveException, MessagingException {
         boolean matchFound = false;
-        String unfoldedAndDecodedHeaderValue = "";
+        String unfoldedAndDecodedHeaderValue = MimeUtility.unfold(header.getValue());
         try {
-            unfoldedAndDecodedHeaderValue =  MimeUtility.decodeText(MimeUtility.unfold(header.getValue()));
-            ZimbraLog.filter.debug("Header value before unfolding and decoding: %s", header.getValue());
+            unfoldedAndDecodedHeaderValue =  MimeUtility.decodeText(unfoldedAndDecodedHeaderValue);
             ZimbraLog.filter.debug("Header value after unfolding and decoding: %s", unfoldedAndDecodedHeaderValue);
         } catch (UnsupportedEncodingException uee) {
-            ZimbraLog.filter.debug("Failed to decode \"%s\"", MimeUtility.unfold(header.getValue()));
```

Comment:
should not remove the log.

Reply:
Yeh, I did think so first, but on second thought, I reconsidered this log may be less informative because of the following thoughts:
(1) because it is a debug log, system admin can't use this log for the production's root cause analysis.
(2) for the debug purpose, the original target header value has been logged before this operation (it is not un-folded though), and the resulting header value is also logged after this operation.  By comparing the original one and the result one, we can recognize what's happened on the decoding process.

That's the reason I removed this log event.